renderdoc/data/glsl/mesh.comp

```diff
@@ -40,37 +40,87 @@ layout(binding = 3, std140) buffer pickresult_buffer
 
 layout (local_size_x = 128, local_size_y = 1, local_size_z = 1) in;
 
+bool TriangleRayIntersect(vec3 A, vec3 B, vec3 C, 
+	vec3 RayPosition, vec3 RayDirection, out vec3 HitPosition)
+{
+	bool Result = false;
+
+	vec3 v0v1 = B - A;
+	vec3 v0v2 = C - A;
+	vec3 pvec = cross(RayDirection, v0v2);
+	float det = dot(v0v1, pvec);
+
+	// if the determinant is negative the triangle is backfacing
+	// if the determinant is close to 0, the ray misses the triangle
+	if (abs(det) > 0.000001f)
```

Comment:
Is it reasonable in this case to ignore backfacing triangles, or should we just detect and flip them so that we can still pick vertices on them? I don't think it's much of a stretch to imagine a situation where e.g. a cliff face model is only intended to be positioned one way in a level, so it isn't watertight and doesn't have a back. If in object space it's facing away from the camera by default, this means people wouldn't be able to pick (m)any vertices until they rotate the object around which might be confusing.

Presumably even if backfaces were included, this would still just pick the 'closer' vert - the ray isn't infinite in length so if you were inside the object it shouldn't pick backfaces behind you in preference to front faces in front of you, say. Is there any other reason to reject backfaces?

Something to consider if it's possible to make it easier to use here, I'm not saying definitively it has to behave one way or another.